OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -1346,17 +1202,194 @@ class AndroidICCAnalysis(val project: SomeProject) extends FPCFAnalysis {
             clone
         }
     }
+
+}
+
+/**
+ * This class is used to reconstruct intent filters from the analysed project. It holds all relevant information to
+ * simulate intent matching.
+ */
+class IntentFilter() {
+    final val addDataAuthority = "addDataAuthority"
+    final val addDataPath = "addDataPath"
+    final val addDataScheme = "addDataScheme"
+    final val addDataSSP = "addDataSchemeSpecificPart"
+    final val addDataType = "addDataType"
+    final val paramRegex = ".*\\(\"|\"\\).*"
+    final val addAction = "addAction"
+    final val addCategory = "addCategory"
+    final val registerReceiver = "registerReceiver"
+    final val receiverRegex = ".*\\: |\\[.*"
+
+    final val filterDataMethods: List[String] = List(
+        addDataAuthority, addDataPath, addDataScheme, addDataSSP, addDataType
+    )
+
+    var receiver: ClassFile = _
+    var componentType: String = _
+    var registeredReceivers: ListBuffer[ObjectType] = ListBuffer.empty[ObjectType]
```

Comment:
It seems weird to have mutable collections in a mutable variable. Usually you'd have either a mutable collection in an immutable val or the other way round.